src/sentry/api/event_search.py

```diff
@@ -584,6 +585,10 @@ def convert_search_filter_to_snuba_query(search_filter):
                 )
             )
         return [name, search_filter.operator, internal_value]
+    elif name == "trace":
+        if search_filter.value.raw_value:
+            return [name, search_filter.operator, uuid.UUID(search_filter.value.raw_value)]
+        return [name, search_filter.operator, search_filter.value.raw_value]
```

Comment:
We should raise an InvalidSearchQuery here. `trace_id` is not nullable in the transactions schema so comparing to `''` is going to fail.